x-pack/plugins/security_solution/public/assistant/helpers.tsx

```diff
@@ -84,30 +84,37 @@ export const augmentMessageCodeBlocks = (
           document.querySelectorAll(`.message-${messageIndex} .euiCodeBlock__controls`)[
             codeBlockIndex
           ],
-        button: sendToTimelineEligibleQueryTypes.includes(codeBlock.type) ? (
-          <SendToTimelineButton
-            asEmptyButton={true}
-            dataProviders={[
-              {
-                id: 'assistant-data-provider',
-                name: `Assistant Query from conversation ${currentConversation.id}`,
-                enabled: true,
-                excluded: false,
-                queryType: codeBlock.type,
-                kqlQuery: codeBlock.content ?? '',
-                queryMatch: {
-                  field: 'host.name',
-                  operator: ':',
-                  value: 'test',
-                },
-                and: [],
-              },
-            ]}
-            keepDataView={true}
-          >
-            <EuiIcon type="timeline" />
-          </SendToTimelineButton>
-        ) : null,
+        button: (
+          <>
+            {sendToTimelineEligibleQueryTypes.includes(codeBlock.type) ? (
+              <SendToTimelineButton
+                asEmptyButton={true}
+                dataProviders={[
+                  {
+                    id: 'assistant-data-provider',
+                    name: `Assistant Query from conversation ${currentConversation.id}`,
+                    enabled: true,
+                    excluded: false,
+                    queryType: codeBlock.type,
+                    kqlQuery: codeBlock.content ?? '',
+                    queryMatch: {
+                      field: 'host.name',
+                      operator: ':',
+                      value: 'test',
+                    },
+                    and: [],
+                  },
+                ]}
+                keepDataView={true}
+              >
+                <EuiIcon type="timeline" />
+              </SendToTimelineButton>
+            ) : null}
+            {DETECTION_RULES_CREATE_FORM_CONVERSATION_ID === currentConversation.title ? (
+              <UpdateQueryInFormButton query={codeBlock.content ?? ''} />
+            ) : null}
```

Comment:
nit: Probably can hide the icon when not on the rule create page as well since it doesn't do anything when clicked if you open the chat on a different page. Depending on the hierarchy (since this isn't a hook), it might not be easy to get the current page, so not a big deal to address this.

Reply:
Also side note: the ref gets stale if you change pages and come back to the create page, select ES|QL rule, then click the update query button from the previous chat (which won't do anything). You have to click the `Ask Assistant` button again and regenerate for functionality to come back. This is a corner case, but just noting the behavior around the ref.

Reply:
@spong 

> Also side note: the ref gets stale if you change pages and come back to the create page, select ES|QL rule, then click the update query button from the previous chat (which won't do anything). You have to click the Ask Assistant button again and regenerate for functionality to come back. This is a corner case, but just noting the behavior around the ref.

I was not able to reproduce this one

https://github.com/user-attachments/assets/f23fb941-a346-4836-937e-1164b914eb5e

Reply:
Hmmm, interesting... 🤔 

That behavior lgtm though, so will resolve this thread 👍